c7n/commands.py

```diff
@@ -417,9 +420,33 @@ def schema_cmd(options):
         resource_mapping = schema.resource_vocabulary(
             cloud_provider)
         components[0] = '%s.%s' % (cloud_provider, components[0])
+    elif components[0] in schema.resource_vocabulary().keys():
+        resource_mapping = schema.resource_vocabulary()
     else:
         resource_mapping = schema.resource_vocabulary('aws')
         components[0] = 'aws.%s' % components[0]
+
+    #
+    # Handle mode
+    #
+    if components[0] == "mode":
+        if len(components) == 1:
+            output = {components[0]: list(resource_mapping[components[0]].keys())}
+            print(yaml.safe_dump(output, default_flow_style=False))
+            return
+
+        if len(components) == 2:
+            if components[1] not in resource_mapping[components[0]]:
+                log.error('{} is not a valid mode'.format(components[1]))
+                sys.exit(1)
+
+            _print_cls_schema(resource_mapping[components[0]][components[1]])
+            return
+
+        # We received too much (e.g. s3.actions.foo.bar)
+        log.error("Invalid selector '{}'. Max of 1 component in the "
+                  "format MODE.TYPE".format(options.resource))
```

Comment:
this reads a little to abstract for a help tool

```
2019-03-30 07:19:41,133: custodian.commands:ERROR Invalid selector 'mode.pull.x'. Max of 1 component in the format MODE.TYPE
```

It would be better as an example of correct usage `eg. custodian schema mode.pull`